src/v/arch/tests/manifest_test.cc

```diff
@@ -0,0 +1,184 @@
+        // use pre-calculated murmur hash value from full ntp path + file name
+        BOOST_REQUIRE_EQUAL(
+          path, "7da10588/test-ns/test-topic/42_0/22-11-v1.log");
+    });
+}
+
+SEASTAR_TEST_CASE(test_empty_manifest_update) {
+    return ss::async([] {
+        arch::manifest m;
+        m.update(make_manifest_stream(empty_manifest_json)).get0();
+        auto path = m.get_manifest_path();
+        BOOST_REQUIRE_EQUAL(
+          path, "a0000000/meta/test-ns/test-topic/42_0/manifest.json");
+    });
+}
+
+SEASTAR_TEST_CASE(test_complete_manifest_update) {
+    return ss::async([] {
+        arch::manifest m;
+        m.update(make_manifest_stream(complete_manifest_json)).get0();
+        auto path = m.get_manifest_path();
+        BOOST_REQUIRE_EQUAL(
+          path, "50000000/meta/test-ns/test-topic/42_1/manifest.json");
+        BOOST_REQUIRE_EQUAL(m.size(), 2);
+        std::map<ss::sstring, arch::manifest::segment_meta> expected = {
+          {"10-1-v1.log",
+           arch::manifest::segment_meta{
+             false, 1024, model::offset(10), model::offset(19)}},
+          {"20-1-v1.log",
+           arch::manifest::segment_meta{
+             false, 2048, model::offset(20), model::offset(29)}}};
+        for (const auto& actual : m) {
+            auto it = expected.find(actual.first);
+            BOOST_REQUIRE(it != expected.end());
+            BOOST_REQUIRE_EQUAL(
+              it->second.base_offset, actual.second.base_offset);
+            BOOST_REQUIRE_EQUAL(
+              it->second.committed_offset, actual.second.committed_offset);
+            BOOST_REQUIRE_EQUAL(
+              it->second.is_compacted, actual.second.is_compacted);
+            BOOST_REQUIRE_EQUAL(
+              it->second.size_bytes, actual.second.size_bytes);
+        }
+    });
+}
+
+SEASTAR_TEST_CASE(test_manifest_serialization) {
+    return ss::async([] {
+        arch::manifest m(manifest_ntp, model::revision_id(0));
+        m.add(
+          arch::segment_name("10-1-v1.log"),
+          {
+            .is_compacted = false,
+            .size_bytes = 1024,
+            .base_offset = model::offset(10),
+            .committed_offset = model::offset(19),
+            .is_deleted_locally = false,
+          });
+        m.add(
+          arch::segment_name("20-1-v1.log"),
+          {
+            .is_compacted = false,
+            .size_bytes = 2048,
+            .base_offset = model::offset(20),
+            .committed_offset = model::offset(29),
+            .is_deleted_locally = false,
+          });
+        auto [is, size] = m.serialize();
+        iobuf buf;
+        auto os = make_iobuf_ref_output_stream(buf);
+        ss::copy(is, os).get();
+
+        auto rstr = make_iobuf_input_stream(std::move(buf));
+        arch::manifest restored;
+        restored.update(std::move(rstr)).get0();
+
+        BOOST_REQUIRE(m == restored);
+    });
+}
+
+SEASTAR_TEST_CASE(test_manifest_difference) {
```

Comment:
I think if you use SEASTAR_THREAD_TEST_CASE you can safely assume your code is running in the context of an ss::async thread, saving you the trouble of having to set it up for every test method

Reply:
Awesome, I'll give it a try.